### ISSUE TITLE
Discrepancies between tests

### ISSUE THREAD

**Issue by Kriskras99** [NationalSecurityAgency/ghidra#2125]
**Describe the bug**
The results of combinedTestReport and parallelCombinedTestReport don't match. In addition integrationTestReport does not produce a report. The combinedTestReport runs 10252 tests (with 274 failures) while the parallelCombinedTestReport runs only 9802 tests (with 103 failures). They also fail different tests:
Test | combinedTestReport | parallelCombinedTestReport
------ | --------------------------- | --------------------------------------
docking.widgets.filechooser | Failed | Passed 
docking.widgets.table.threaded  | Passed | Failed
ghidra.app.plugin.core.compositeeditor | Passed | Failed
ghidra.app.plugin.core.data | Passed | Failed
ghidra.app.plugin.core.datamgr.editor | Passed | Failed
ghidra.app.plugin.core.functiongraph | Passed | Failed
ghidra.app.plugin.core.functionwindow | Passed | Failed
ghidra.app.plugin.core.memory | Passed | Failed
ghidra.app.plugin.core.navigation | Passed | Failed
ghidra.app.plugin.core.progmgr | Passed | Failed
ghidra.app.plugin.core.script | Failed | Passed
ghidra.app.plugin.core.symboltree | Passed | Failed
ghidra.feature.vt.api | Failed | Missing
ghidra.feature.vt.api.correlator.address | Failed | Missing
ghidra.feature.vt.api.markupitem | Failed | Passed
ghidra.feature.vt.gui.provider | Failed | Missing
ghidra.feature.vt.gui.provider.functionassociation | Failed | Missing
ghidra.framework.main.datatree | Passed | Failed
ghidra.framework.plugintool.dialog | Passed | Failed
ghidra.framework.project.tool | Failed 2 | Failed 1
ghidra.graph.viewer.edge | Failed | Passed

_Note that in this table are only tests that failed in one of the runs, if a test passed in one and is missing in the other it is not in this table._
If tests where only failing extra on the parallel side, this could be blamed on running in parallel (and might still be for some tests). But this is not what is happening.

**To Reproduce**
1. gradle unitTestReport -PreportDir=../../unitTest 
2. gradle integrationTest -PreportDir=../../integrationTest
3. gradle combinedTestReport -PreportDir=../../combinedTest
4. gradle parallelCombinedTestReport -PreportDir=../../parallelTest
5. Compare results of combinedTest and parallelCombinedTest
6. See that the integrationTest does not even produce a report

**Expected behavior**
Have the same amount of tests run and fail. Also have a test report from the integrationTest

**Attachments**
I've attached the test reports for unitTest, combinedTest and parallelCombinedTest
[combinedTestReport.zip](https://github.com/NationalSecurityAgency/ghidra/files/4953464/combinedTestReport.zip)
[parallelTestReport.zip](https://github.com/NationalSecurityAgency/ghidra/files/4953465/parallelTestReport.zip)
[unitTestReport.zip](https://github.com/NationalSecurityAgency/ghidra/files/4953467/unitTestReport.zip)

**Environment (please complete the following information):**
 - OS: Arch Linux (kernel 5.7.9)
 - Java Version: 11.0.8
 - Ghidra Version: Master (ac0a4c9)

**Additional context**
Test | Tests | Failures | Ignored | Duration | Percentage
----- | ------- | ----------- | ---------- | ----------- | ----------------
unitTest | 3735 | 0 | 0 | 12m18.58s | 100%
combinedTest | 10252 | 274 | 0 | 1h42m22.61s | 97%
parallelCombinedTest | 9802 | 103 | 0 | 4h52m57.23s | 98%

_Please note that the duration given is inaccurate, unitTest takes about 20m, combinedTest 2h45m, parallelCombinedTest not sure_

**Comment by dragonmacher** [comment]
For now, the parallel tests are considered to be a work-in-progress.     Running the tests in parallel exposes latent timing issues that do not often appear in the serial tests.

I am closing this ticket for now, as the work to fix this is ongoing, is low priority and will be a long-term project.
